nettests/nettests.go

```diff
@@ -273,6 +279,117 @@ func (c *Controller) Init(nt *mk.Nettest) error {
 	return nil
 }
 
+// Run runs the selected nettest using the related experiment
+// with the specified inputs.
+//
+// This function will continue to run in most cases but will
+// immediately halt if something's wrong with the file system.
+//
+// This is the codepath for running ooni/probe-engine nettests.
+func (c *Controller) Run(exp *experiment.Experiment, inputs []string) error {
+	ctx := context.Background()
+
+	// This will configure the controller as handler for the callbacks
+	// called by ooni/probe-engine/experiment.Experiment.
+	exp.Callbacks = handler.Callbacks(c)
+
+	c.msmts = make(map[int64]*database.Measurement)
+
+	// These values are shared by every measurement
+	reportID := sql.NullString{String: "", Valid: false}
+	resultID := c.res.ID
+
+	log.Debug(color.RedString("status.queued"))
+	log.Debug(color.RedString("status.started"))
+	log.Debugf("OutputPath: %s", c.msmtPath)
+
+	// TODO(bassosimone): double check that we're passing the right options
+	// to MK? We may be passing less option than needed currently, even though
+	// the code works for the common configuration.
```

Comment:
Note to self: this should either be addressed in this PR or formalised as an issue.